.github/workflows/check.yaml

```diff
@@ -0,0 +1,29 @@
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+      - master
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref || '' }}-${{ github.base_ref || '' }}
```

Comment:
I'm using a slightly different setting in my repos now, to make sure that we build all commits on the main branch. Can be changed in a forthcoming PR.